pkg/sanity/controller.go

```diff
@@ -1110,19 +1117,21 @@ var _ = DescribeSanity("Controller Service [Controller Server]", func(sc *TestCo
 			// ControllerPublishVolume
 			By("calling controllerpublish on that volume")
 
-			conpubvol, err := c.ControllerPublishVolume(
-				context.Background(),
-				&csi.ControllerPublishVolumeRequest{
-					VolumeId:         vol.GetVolume().GetVolumeId(),
-					NodeId:           ni.GetNodeId(),
-					VolumeCapability: TestVolumeCapabilityWithAccessType(sc, csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER),
-					Readonly:         false,
-					Secrets:          sc.Secrets.ControllerPublishVolumeSecret,
-				},
-			)
-			Expect(err).NotTo(HaveOccurred())
+			for i := 0; i < idempotentcount; i++ {
```

Comment:
This modifies the existing "ControllerPublishVolume should return appropriate values (no optional values added)" test. Idempotency of the other controller functions isn't covered.

Can you use the same pattern as for the node testing where a helper function runs the full test such that *every* function is repeated n times? The simple version of the test then only does it once.

Reply:
> same pattern as for the node

good point, that is a cleaner way.
While looking for tests that can be repeated, I notice that we have two code blocks
that are functionally identical, only differing by few strings (volume name and By), and minor ordering of statements (not meaningful functionally).
One block is lines 1068..1152  (master branch)
another is 1385..1470
The idea (by names) seems to be that first one tests Publish and 2nd one Unpublish,
but as both perform Create-Publish-Unpublish-Delete, these both test same thing.
If we really want to have such repeated test group, it makes sense to turn this code into function
(which I BTW do anyway in repetition-enabling change).
But we can also ask question, can we drop the 2nd instance, as it does not add anything new,
as both Publish and Unpublish do get tested same way already in 1st instance.